src/TestingUtils/Microsoft.AspNetCore.Testing/test/MaximumOSVersionTest.cs

```diff
@@ -42,7 +42,7 @@ public void RunTest_Win10_RS4()
             Assert.NotNull(versionKey);
             var currentVersion = (string)versionKey.GetValue("CurrentBuildNumber");
             Assert.NotNull(currentVersion);
-            Assert.True(17134 >= int.Parse(currentVersion));
+            Assert.True(17134 >= int.Parse(currentVersion), $"Unexpected build number {currentVersion} on {Environment.OSVersion}.");
```

Comment:
@MattGal this displays very strange information when the assertion fails:
```
Unexpected build number 17763 on Microsoft Windows NT 6.2.9200.0.
```

But, according to https://en.wikipedia.org/wiki/List_of_Microsoft_Windows_versions, `17763` is the build number of the latest Windows Server 2019 release. But, `6.2.9200.0` indicates the agent is running Windows Server 2012 (not R2) and `9200` is the build number of the latest release there.

Reply:
FYI I need to revise our testing attributes to handle server machines and get them working in the new world. Right now they seem to work as expected only for the client OS.

Reply:
Tagging @Tratcher

Reply:
`Environment.OSVersion` lies for back-compat reasons unless you have an app manifest. VS and dotnet test both have manifests so that's not usually a problem?

What version did this queue use to run on?

Maybe server 2019 has a new manifest guid?
https://docs.microsoft.com/en-us/windows/win32/sbscs/application-manifests

Reply:
What Tratcher said.  From our perspective though, whenever we say server we are aiming for "latest non-core fully patched server as of when the image was generated" now.

Reply:
Being current is great, we just need to figure out why our version detection broke.

Reply:
Hmm, the app manifest docs only go up to Server 2016, I can't find any information about 2019.

Reply:
The owners of the app manifest docs claim the GUID is unchanged for server 2019. Might have to set up some machines and confirm.
https://github.com/MicrosoftDocs/feedback/issues/2477#issuecomment-594084535

Reply:
If it used the same GUID, these tests wouldn't have failed and we wouldn't see `6.2.9200.0`